### PR TITLE
Increase journald rate limiter when running test-end-to-end-docker.sh

### LINKED ISSUE #12558: test/end-to-end/core.sh:376: executing 'oc logs kubectl-with-token' expecting success and text 'Using in-cluster configuration': the output content test failed
Seen in https://ci.openshift.redhat.com/jenkins/job/test_pull_requests_origin_integration/10167/consoleFull:

```
FAILURE after 0.223s: test/end-to-end/core.sh:375: executing 'oc logs kubectl-with-token' expecting success and text 'Using in-cluster configuration': the output content test failed
Standard output from the command:
NAME                 READY     STATUS    RESTARTS   AGE
kubectl-with-token   1/1       Running   0          5s
```

### PR DESCRIPTION
**Pull request by soltysh**
Fixes #12558 by increasing rate limiter values while running test-end-to-end-docker.sh. Additionally, this re-enables the tests that were disabled due to that error.

@stevekuznetsov another try... 
@mfojtik fyi

### CODE REVIEW

**Comment by soltysh** [comment]
[test]

**Comment by soltysh** [comment]
I've removed that offending deprecation warning using the recommended/new approach for `oc exec`.

**Comment by soltysh** [comment]
Flake https://github.com/openshift/origin/issues/13757

[test]

**Comment by soltysh** [comment]
@ncdc I'm baffled by this error. The case is when I increase `RateLimitInterval` and `RateLimitBurst` in journald.conf and get to `oc rsh dc/docker-registry cat config.yml` the end-to-end test will hang waiting for the connection. Looking closer at the instance I've noticed that generally docker will hang after some time. Usually, only restart will help. Any ideas/thoughts what might be causing this? Pointers where should I look for it? I can't reproduce this manually when running `oc cluster up` and trying `oc rsh`.

**Comment by ncdc** [comment]
I'm off today but will look either this weekend or Monday.

**Comment by soltysh** [comment]
Sure, sure, no rush :)

**Comment by ncdc** [comment]
I wonder if we're starting to run into https://github.com/kubernetes/kubernetes/pull/43922

**Comment by soltysh** [comment]
I'll try to spin up a new env with that fix in, I'm hitting this quite consistently with this PR.

**Comment by ncdc** [comment]
I'm doubtful that's the issue but we'll see.

**Comment by soltysh** [comment]
Nope, that PR didn't help with the problem 😢 My last resort is the current rebase, I guess.

**Comment by ncdc** [comment]
Can you get stack dumps from openshift and oc when this happens?

**Comment by soltysh** [comment]
Nope, the rebase does not help here, either. Sigh...

@smarterclayton do you have any ideas why `oc rsh dc/docker-registry cat config.yml` in end-to-end test will hang after increasing `RateLimitInterval` and `RateLimitBurst` in `journald.conf`? I've tried manually changing those values and running `oc rsh` and it worked perfectly, but for some reason it will fail when part of end-to-end.

**Comment by ncdc** [comment]
I really think we need stack dumps from openshift, oc, and possibly docker
when this happens. I don't remember the name of the environment variable
(maybe DEBUG?), but it's possible to enable the pprof http endpoint for
docker just like we have in openshift.

**Comment by smarterclayton** [comment]
Is there something waiting for a newline in the output?  Docker hang or
origin hang?

**Comment by soltysh** [comment]
The symptoms are as follows, the end-to-end will hang (not reacting to any sort of input), openshift seems to be working correctly. After some time docker daemon will hang (not reacting to any docker commands), and only restarting the daemon helps with it. I've collected some stack dumps and I'll be going through them today, with hope to find something useful.

**Comment by ncdc** [comment]
If you can reliably reproduce, make sure you talk to @jwhonce @nalind

**Comment by smarterclayton** [comment]
Getting stable tests is p0.

**Comment by soltysh** [comment]
It looks like the problem previously was that docker does not react well to restarting journald, so I've added restarting docker after restarting journald, as well. Seems to be working, let's give it a try. 
/me crosses fingers
[test]

**Comment by soltysh** [comment]
Fixed package typo in UPSTEAM cherry-pick commit. Waiting for green tests.

**Comment by soltysh** [comment]
Flake https://github.com/openshift/origin/issues/14434.
[test]

**Comment by smarterclayton** [comment]
LGTM

File: hack/test-end-to-end-docker.sh [hunk 1]
```diff
@@ -21,12 +21,32 @@ fi
...
 	return_code=$?
 	os::cleanup::all "${return_code}"
+
+	# restore journald to previous form
+	os::log::info "Restoring journald limits"
+	${USE_SUDO:+sudo} mv /etc/systemd/{journald.conf.bak,journald.conf}
+	${USE_SUDO:+sudo} systemctl restart systemd-journald.service
+	# Docker has "some" problems when journald is restarted, so we need to
+	# restart docker, as well.
+	${USE_SUDO:+sudo} systemctl restart docker.service
```
Comment (smarterclayton) [inline hack/test-end-to-end-docker.sh]: This is fairly invasive.  But I don't have a better place for it.

Also, please ensure that you're actually on a system with systemctl, i.e. macs.

**Comment by smarterclayton** [comment]
Removing LGTM, this breaks macs.

File: hack/test-end-to-end-docker.sh (hunk 1 above)
Comment (soltysh) [inline hack/test-end-to-end-docker.sh]: Should be fixed, I've added `os::util::ensure::system_binary_exists 'systemctl'` checks around these calls.

**Comment by soltysh** [comment]
Flke https://github.com/openshift/origin/issues/10773.

**Comment by smarterclayton** [comment]
LGTM

**Comment by smarterclayton** [comment]
Going to the head of the line

**Comment by smarterclayton** [comment]
Does not conflict with the PR being merged right now.  Moving to the head of the line to fix flakes.

**Comment by stevekuznetsov** [comment]
Nice

**Comment by soltysh** [comment]
img

**Comment by mfojtik** [comment]
\o/

**Pull request merged**